docs/high/datatype.rst

```diff
@@ -0,0 +1,47 @@
+.. _datatype:
+
+Datatypes
+=========
+
+Usually, :class:`Datatype` objects are created for you on the fly and you do not
+have to worry about them. However, in certain cases, namely when no HDF5
+equivalent exists for a given :class:`dtype`, you must register the :class:`dtype`
+manually for use with h5py.::
+
+    arr = np.array([np.datetime64('2019-06-30')])
+    h5py.register_dtype(arr.dtype)
+    dset = f.create_dataset("datetimes", data=arr)
+
+.. note::
+
+    It is important to notice that the types registered in this way will only
+    be readable be NumPy and compatible tools, and that this format may not be
+    universally accepted. In the case of a third party ``dtype``, it will only
+    be readable insofar as the ``dtype`` is binary-compatible across old
+    versions. In general, opaque datatypes are very sensitive to how you may
+    decide to encode your data.
+
+Reference
+=========
+
+.. class:: Datatype(identifier)
+
+    Datatype objects are typically created via :func:`register_dtype` function,
```

Comment:
This isn't quite right - `register_dtype` doesn't return a `Datatype` object, or even the lower-level `TypeID`.